src/utils/index.ts

```diff
@@ -248,6 +264,150 @@ export interface ICurrentStats {
+  );
+
+  return {
+    ubq: ubqHoldings,
+    ink: inkHoldings,
+    esch: eschHoldings,
+    lp: {
+      ubqInk: ubqInkHoldings,
+      gransInk: gransInkHoldings,
+      inkEsch: inkEschHoldings,
+    },
+  } as IDaoHoldings;
+};
+
+// Calculates the circulating INK supply based on the amount still held by
+// the DAO minting address plus a sum of all of the individual farm holdings
+export const getCirculatingSupply = async (provider: provider): Promise<ICirculatingSupply> => {
+  const totalSupply = new BigNumber(88 * 1000 * 1000); // 88 million
+
+  // This address holds the minted INK which is yet to be distributed
+  const INKMINTEDHOLDINGADDRESS = "0xB47D5874D2db5f398cfA0E53a5A020362F2AEAeF";
+
+  // Grab all the individual INK quantities
+  let heldByDeployer = await getBalanceAsBigNum(provider, INK, INKMINTEDHOLDINGADDRESS);
+  let heldByUBQINK = await getBalanceAsBigNum(provider, INK, INK_UBQ_FarmContract);
+  let heldByGRANSUBQ = await getBalanceAsBigNum(provider, INK, INK_GRANS_FarmContract);
+  let heldByUBQESCH = await getBalanceAsBigNum(provider, INK, INK_ESCH_FarmContract);
+  const subtotal = heldByDeployer.plus(heldByUBQINK).plus(heldByGRANSUBQ).plus(heldByUBQESCH).toString();
+
+  // Total them up, and subtract them from the totalSupply
+  let circulatingTotal = totalSupply.minus(subtotal);
+
+  // console.log(
+  //   "heldByDeployer:",
+  //   heldByDeployer,
+  //   "heldByUBQINK:",
+  //   heldByUBQINK,
+  //   "heldByGRANSUBQ:",
+  //   heldByGRANSUBQ,
+  //   "heldByUBQESCH:",
+  //   heldByUBQESCH,
+  //   "subtotal:",
+  //   subtotal,
+  //   "total:",
+  //   circulatingTotal.toString()
+  // );
+
+  return {
+    heldByMultisig: heldByDeployer,
+    heldByUBQINK: heldByUBQINK,
+    heldByGRANSUBQ: heldByGRANSUBQ,
+    heldByUBQESCH: heldByUBQESCH,
+    total: circulatingTotal,
+  } as ICirculatingSupply;
+};
+
+export interface IDailyTransactions {
+  count: number;
+}
+
+export const getDailyTransactions = async (provider: provider): Promise<IDailyTransactions> => {
+  if (provider === undefined || provider === null) {
+    return { count: 0 };
+  }
+  const web3 = new Web3(provider);
+
+  const oneDayInSeconds = 60 * 60 * 24;
+  const blockTime = 88;
+  const oneDayInBlocks = Math.floor(oneDayInSeconds / blockTime);
+  let inkResults = [];
+  let inkUbqFarmResults = [];
+  let gransInkFarmResults = [];
+  let inkEschFarmResults = [];
+  const currentBlock = await web3.eth.getBlockNumber();
+
+  try {
+    // TODO: Perhaps we should just store these in available farms?
+    const INKCONTRACT = getERC20Contract(provider, INK);
+    const UBQINKFARMCONTRACT = getFarmContract(provider, INK_UBQ_FarmContract);
+    const GRANSINKFARMCONTRACT = getFarmContract(provider, INK_GRANS_FarmContract);
+    const INKESCHFARMCONTRACT = getFarmContract(provider, INK_ESCH_FarmContract);
+
+    // TODO: any way to make this more flexible using availablefarms?
```

Comment:
this can easily be an array that uses foreach, like the other suggestion.

Reply:
fixed

Reply:
issue created